device_backends/include/NumericAddressedLowLevelTransferElement.h

```diff
@@ -124,16 +122,24 @@ namespace ChimeraTK {
    protected:
     /** Set the start address (inside the bar given in the constructor) and number
      * of words of this accessor. */
-    void setAddress(size_t startAddress, size_t numberOfWords) {
+    void setAddress(size_t startAddress, size_t numberOfBytes) {
       // change address
       _startAddress = startAddress;
-      _numberOfWords = numberOfWords;
-
-      // allocated the buffer
-      rawDataBuffer.resize(_numberOfWords);
+      _numberOfBytes = numberOfBytes;
+
+      // make sure access is properly aligned
+      auto alignment = _dev->minimumTransferAlignment();
+      auto start_padding = _startAddress % alignment;
```

Comment:
1. I don't see where the padding is used when filling the rawDataBuffer from the user buffer, or the other way around.
2. When writing, the padded bytes are also written, which might change content on the device. A read/modify/write is needed.

Reply:
To 1.: The access is always done through `begin()` which takes the actual start address of the high level accessor as an argument. That way we can have any internal start address of the low level transfer element, as long as the accessed region is covered. It does not matter if the start address is move because two transfers are merged or to fulfil the alignment requirement - the same mechanism is used.

To 2.: Correct, I will fix this.

Reply:
2. is fixed now with 010bf34b5daa48f938b5b3875b5de8309b15a2d1

Reply:
I don't know if we need the lock:
Accessors are not thead-safe, and I think also TransferGrous and LowLevelTransferElements are not. So from that thread there is no concurrency.
From another thread with a different accessor and a different transfer element it not effective (never is, cannot be conceptually)

Reply:
I would have to read the spec again, but I think the low level transfer element is not allowed to read from the hardware in preRead(). But at the moment I see no other way.
Is it ensured that the low level pre-read is executed before the high-level pre-read?

Reply:
Accessors are thread-safe to each other. We are allowed to use different accessors from different threads. Since the padding bytes might be data bytes to another accessor, we need this Backend-global (!) lock.

I have not found anything in the spec which says that. I think we discussed this in the past and decided it must be allowed, because it might need to communicate with the device to decide whether to throw a logic_error.

Do you maybe mean `preWrite()` instead of `preRead()`? The order of `preWrite()` is defined in `NumericAddressedBackendRegisterAccessor::doPreWrite()`. I have put a comment there how it has to be. Since the low level transfer element is not used anywhere else (muxed accessor directly calls read/write of the device), this is sufficient.

Reply:
> access

:+1: I did not see that it was a deferred lock to a class-wide mutex.